akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2Multiplexer.scala

```diff
@@ -134,6 +133,14 @@ private[http2] trait Http2MultiplexerSupport { logic: GraphStageLogic with Stage
 
       private val controlFrameBuffer: mutable.Queue[FrameEvent] = new mutable.Queue[FrameEvent]
       private val sendableOutstreams: mutable.Queue[Int] = new mutable.Queue[Int]
+      def enqueueOutstream(streamId: Int): Unit = {
+        if (isDebugEnabled) require(!sendableOutstreams.contains(streamId), s"Stream [$streamId] was enqueued multiple times.") // requires expensive scanning -> avoid in production
```

Comment:
hmm, people might have debug enabled in production - but in that case the scanning is perhaps also fine-ish?

Reply:
With the amount of logging going on, that would be a problem anyway, especially for busy connections with lots of streams.